idaes/gas_solid_contactors/properties/methane_iron_OC_reduction/solid_phase_thermo.py

```diff
@@ -585,3 +596,10 @@ def default_material_balance_type(blk):
 
     def default_energy_balance_type(blk):
         return EnergyBalanceType.enthalpyTotal
+
+    def calculate_scaling_factors(self):
+        super().calculate_scaling_factors()
+
+        if hasattr(self, "sum_component_eqn"):
+            for t, v in self.sum_component_eqn.items():
+                iscale.set_scaling_factor(v, 1e2)
```

Comment:
Some of the component mole fractions may become very small, either through inlet specifications (see the MB flowsheet example where the inlet water fraction is 1E-5) or if methane approaches full conversion by the reactor outlet. The scaling helps address convergence issues; @chineduobiora may be able to expand upon the reasoning behind the scaling factor.

Reply:
Assuming this is a sum of mole fractions, you should try to get the currently defined scaling factor first, with a fall back to a default if one has not been defined.

Reply:
I have no real problem with this scaling factor, but my intuition is that adding a scaling factor of 1e2 to a constraint actually makes it _harder_ to converge. I.e., we need to converge our mole fraction variables to a tighter tolerance to satisfy this constraint to an absolute tolerance of 1e-8. This factor also increases the constraint's Jacobian entries from 1 to 100, which we may or may not want? 

Again no real problem here, I'm just being difficult because I'm trying to arrive at a more systematic understanding of scaling myself and am always curious about hard-coded scaling factors.

Reply:
1e2 was originally hardcoded here because my experience is to have the mole fraction variables scaled by at least a factor of 100. For the generic application I would lean towards @andrewlee94's suggestion to scale this based on a default scaling factor (using the default scaling for the mole fraction variables is a good choice). However, to @Robbybp's point the "right" scaling factor also tends to be problem specific, and a user should be able to adjust scaling factors from the flowsheet level as needed.

Reply:
@bpaul4 the gas proppack we discussed at our last meeting has examples of how default scaling can be set for individual variables (see the end of the PhysicalParameterData Class), and how they can be applied to scale constraints.